lib/ansible_tower_client/job_template.rb

```diff
@@ -10,11 +10,15 @@ def initialize(api, raw_body)
       super
     end
 
-    def launch(vars = {})
+    def launch(options = {})
       launch_url = "#{url}launch/"
-      extra = JSONValues.new(vars).extra_vars
-      resp = api.post(launch_url, extra).body
-      job = JSON.parse(resp)
+      new_limit = options.dup.delete('limit')
```

Comment:
In this implementation, we still pass limit in the post options on line 17.  That was supposed to be removed from the options since the block already takes care of setting the limit for us.

```ruby
irb(main):001:0> raw_options = {:limit => "abc", :extra_vars => "xyz"}
=> {:limit=>"abc", :extra_vars=>"xyz"}
irb(main):002:0> options = raw_options
=> {:limit=>"abc", :extra_vars=>"xyz"}
irb(main):003:0> raw_options.object_id
=> 47375013423080
irb(main):004:0> options.object_id
=> 47375013423080
irb(main):005:0> limit = options.dup.delete(:limit)
=> "abc"
irb(main):006:0> options
=> {:limit=>"abc", :extra_vars=>"xyz"}
```

The `dup` needs to be on its own line because:
```ruby
irb(main):001:0> raw_options = {:limit => "abc", :extra_vars => "xyz"}
=> {:limit=>"abc", :extra_vars=>"xyz"}
irb(main):002:0> options = raw_options.dup
=> {:limit=>"abc", :extra_vars=>"xyz"}
irb(main):003:0> raw_options.object_id
=> 47002009217800
irb(main):004:0> options.object_id
=> 47002008338660
irb(main):005:0> limit = options.delete(:limit)
=> "abc"
irb(main):006:0> limit
=> "abc"
irb(main):007:0> options
=> {:extra_vars=>"xyz"}
irb(main):008:0> raw_options
=> {:limit=>"abc", :extra_vars=>"xyz"}
```